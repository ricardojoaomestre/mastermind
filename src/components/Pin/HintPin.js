import styled from 'styled-components';
import BasePin from './BasePin';

const StyledHintPin = styled(BasePin)`
    width: 14px;
    height: 14px;
    border-radius: 8px;
    border: 1px solid #C4C4C4;
    background-color: ${props => props.value === "correct" ? "#000000" : props.value === "semicorrect" ? "#D9DADC" : "transparent"};

    &:hover {
        cursor: default;
    }
`;

const HintPin =  (props) => <StyledHintPin {...props}/>;

export default HintPin;