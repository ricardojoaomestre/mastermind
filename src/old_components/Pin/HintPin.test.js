import { render, cleanup } from "@testing-library/react";

import HintPin from "./HintPin";

afterEach(cleanup);

describe("HintPin", () => {
  it("must render as expected", () => {
    const { container } = render(<HintPin />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
