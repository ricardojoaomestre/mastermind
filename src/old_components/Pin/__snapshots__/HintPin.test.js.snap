// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintPin must render as expected 1`] = `
<div
  class="sc-bdfBQB sc-gsTEea ioqCIK iBEhSO"
/>
`;
